### PR TITLE
[luci-interpreter] Support FloorMod kernel on Linux

### PR DESCRIPTION
**Pull request by ragmani**
This supports fp32 FloorMod kernel on Linux.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

related to #10659

### CODE REVIEW

**Comment by seanshpark** [comment]
Plz fix format check failure

**Comment by ragmani** [comment]
I revised `compiler/luci-value-test/test.lst` as well.

File: compiler/luci-interpreter/pal/linux/KernelsToBuild.lst [hunk 1]
```diff
@@ -17,6 +17,7 @@ REGISTER_KERNEL(ExpandDims)
 REGISTER_KERNEL(Fill)
 REGISTER_KERNEL(Floor)
 REGISTER_KERNEL(FloorDiv)
+REGISTER_KERNEL(FloorMod)
```
Comment (seanshpark) [inline compiler/luci-interpreter/pal/linux/KernelsToBuild.lst:20]: plz split this to another PR.
`compiler` has policy not to mix changes of `compiler/*` in one PR.

File: compiler/luci-value-test/test.lst [hunk 2]
```diff
@@ -54,8 +54,8 @@ addeval(Exp_000)
 addeval(Floor_000)
 addeval(FloorDiv_000)
 addeval(FloorDiv_001)
-#addeval(FloorMod_000)
-#addeval(FloorMod_001)
+addeval(FloorMod_000)
+addeval(FloorMod_001)
```
Comment (seanshpark) [inline compiler/luci-value-test/test.lst:58]: plz split this to another PR.
`compiler` has policy not to mix changes of `compiler/*` in one PR.

Comment (ragmani) [reply compiler/luci-value-test/test.lst:58]: OK. I will.

File: compiler/luci-interpreter/src/kernels/FloorMod.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  };
+
+  const auto x_data = getTensorData<float>(x());
+  const auto y_data = getTensorData<float>(y());
+
+  // Check the denominator
+  for (int i = 0; i < getTensorShape(y()).FlatSize(); ++i)
+  {
+    LUCI_INTERPRETER_CHECK(y_data[i] != 0);
+  }
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/FloorMod.cpp:67]: This behavior is different from TFLite. Divide-by-zero is checked only when y is int32 or int64. For fp32, divide-by-zero is allowed (the tensor may contain NaN values).

Comment (ragmani) [reply compiler/luci-interpreter/src/kernels/FloorMod.cpp:67]: It's not hard to remove this due to difference from TFLite, but I'm confused.
`FloorDiv` kernel of `luci-interpreter` has already checked this for fp32 and even has a test. Does `FloorDiv` kernel also need to be modified?
Please correct me if I understood it wrong.

Comment (jinevening) [reply compiler/luci-interpreter/src/kernels/FloorMod.cpp:67]: Hmm.. It seems that TFLite checks divide-by-zero for fp32 in `FloorDiv` (but not in `FloorMod`). I don't know why TFLite team handles `FloorDiv` and `FloorMod` differently. 🤔

Comment (ragmani) [reply compiler/luci-interpreter/src/kernels/FloorMod.cpp:67]: OK, I will modify it to be the same as TFLite.

**Comment by ragmani** [comment]
I added support for integer types as well.

File: compiler/luci-interpreter/src/kernels/FloorMod.test.cpp [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+{
+  Tensor input1_tensor = makeInputTensor<DataType::FLOAT32>({1}, {1.f}, _memory_manager.get());
+  Tensor input2_tensor = makeInputTensor<DataType::FLOAT32>({1}, {1.f}, _memory_manager.get());
+  Tensor output_tensor = makeOutputTensor(DataType::S8);
+
+  FloorMod kernel(&input1_tensor, &input2_tensor, &output_tensor);
+  EXPECT_ANY_THROW(kernel.configure());
+}
+
+TEST_F(FloorModTest, Input_Type_Mismatch_NEG)
```
Comment (seanshpark) [inline compiler/luci-interpreter/src/kernels/FloorMod.test.cpp:346]: Can you please add num_of(NEG test case) >= num_of(POS test case) ?

Comment (ragmani) [reply compiler/luci-interpreter/src/kernels/FloorMod.test.cpp:346]: Yes, I will.

Comment (ragmani) [reply compiler/luci-interpreter/src/kernels/FloorMod.test.cpp:380]: I added NEG tests.

File: compiler/luci-interpreter/src/kernels/FloorMod.cpp (hunk 3 above)
Comment (jinevening) [reply compiler/luci-interpreter/src/kernels/FloorMod.cpp:90]: I guess you forgot to remove the check in fp32?

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

File: compiler/luci-interpreter/src/kernels/FloorMod.cpp (hunk 3 above)
Comment (ragmani) [reply compiler/luci-interpreter/src/kernels/FloorMod.cpp:90]: Oops. Sorry. I will remove it.

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by jinevening — APPROVED** [review]
LGTM Thank you
